### ISSUE TITLE
successfully sign in to NU shibboleth

### ISSUE THREAD

**Issue by lmkw** [NUARIG/competitions#438]
- [x] Upgrade Competitions code to use Ruby 2.6.6.
- [x] Add a new local version of competitions for deploying to the new environment
- [x] Deploy application to new server for competitions-staging.fsm.northwestern.edu
- [x] Add necessary configuration files to the server.
- [x] Add metadata configurations from NUIT's IDP to our application.
- [x] Send our metadata to NUIT. 

All of this will require manual testing for each step also.

**Comment by lmkw** [comment]
depends on #434, #435, #437,

**Comment by mattbaumann1** [comment]
We have applied for Shibboleth, see ticket #437. After discussion with XiaoXia Dong, it was determined that we wanted to apply for Shibboleth rather than NU's SSO Passport. I am renaming the ticket from "successfully sign in to NU passport using SSO" to reflect the new process.

**Comment by lmkw** [comment]
Still waiting for server approval...

**Comment by mattbaumann1** [comment]
@firaswehbe, I believe you are getting the notifications for this server. Any updates?

**Comment by firaswehbe** [comment]
I can't find anything in my inbox. Can you give me a string to search by? or ticket number?

**Comment by mattbaumann1** [comment]
@firaswehbe, I don't have a ticket number. @nutodd has Rocky looking into it.

**Comment by nutodd** [comment]
Footprints ticket# 991642
Recent questions about this include: What URL will be used?
Are we requesting a new server because we want a clean, isolated space for app/db?

**Comment by mattbaumann1** [comment]
Currently blocked by #512 .

**Comment by mattbaumann1** [comment]
The upgrade to Ruby 2.6.6 will be handled in the open source branch in ticket #438 . This will make deployment to the new NU servers easier and it is the current stable version of 2.6. Each of the servers will need to be upgraded and this will be handled in the following tickets. See above.

**Comment by mattbaumann1** [comment]
I was able to run through the full deployment script today. There are still errors in the log and I am still working with Xiaoxia on the SSO configurations.

**Comment by mattbaumann1** [comment]
The page now loads. We still need to get the configs set for the SSO.

**Comment by mattbaumann1** [comment]
We are waiting on XiaoXia to work through the config issues with the SSO.

**Comment by mattbaumann1** [comment]
I had a meeting with XiaoXia this morning. We went through the configuration values and most seem better. However, we are still getting the following error with the new SSO URL. 

Northwestern University - Stale Request
You may be seeing this page because you used the Back button while browsing a secure web site or application. Alternatively, you may have mistakenly bookmarked the web login form instead of the actual web site you wanted to bookmark or used a link created by somebody else who made the same mistake.
 Left unchecked, this can cause errors on some browsers or result in you returning to the web site you tried to leave, so this page is presented instead.

**Comment by mattbaumann1** [comment]
From XiaoXia on Friday, "This is an known issue on this new version and we are working on resolving this. Thanks."

**Comment by mattbaumann1** [comment]
XiaoXia informed us that there was a problem with our XML metadata. She received the following error. 

2020-06-11 13:42:00,388 - ERROR [org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver:462] - Metadata Resolver FilesystemMetadataResolver competitionStageMetadata: Unable to unmarshall metadata: net.shibboleth.utilities.java.support.xml.XMLParserException: Unable to parse inputstream, it contained invalid XML

We cut and pasted the page source for (https://competitions-cd2h-staging.fsm.northwestern.edu/users/saml/metadata) into two different XML validators (ex. https://www.samltool.com/validate_xml.php). When viewing the page source as “metadata,” we get the message, “the XML is valid.” When viewing it as “AuthNRequest,” it comes back with the error message, “invalid XSD for this XML.”

**Comment by mattbaumann1** [comment]
Advances in the since the last update. Through comparisons with the requests and responses created when signing into the SSO passport system. It became clear that the name_identifier_format: needed to be set to transient. This allowed us to receive a SAMLResponse from the SSO passport system instead of an error. 

Once we got the response, we were able to see the way the attributes were being passed. This allowed us to change the attribute-map. It isn't clear, but we think this is set up correctly currently. 

Finally, today we were able to determine that there is a problem with the creation of the @document for the SAMLResponse and this seems like it might be due to the lack of a signature/certificate in the SAMLResponse from SSO Passport. In comparison both OKTA and UNA contain the certificate in their responses.
